File: python_programming_basics/nested_loops/l_04_sum_of_two_numbers.py
start_num = int(input())
end_num = int(input())
magic_num = int(input())
counter = 0
is_equal = False

for first_combination in range(start_num, end_num + 1):
    for second_combination in range(start_num, end_num + 1):
        counter += 1

        if first_combination + second_combination == magic_num:
            is_equal = True
            break

    if is_equal:
        break

if is_equal:
    print(f"Combination N:{counter} ({first_combination} + {second_combination} = {magic_num})")

else:
    print(f"{counter} combinations - neither equals {magic_num}")

File: JavaScript Basics/06. Nested Loops/l-04-Sum-of-Two-Numbers.js
function findMagicCombination(start, end, magicNumber) {
    let combinationCount = 0;

    for (let i = start; i <= end; i++) {
        for (let j = start; j <= end; j++) {
            combinationCount++;
            if (i + j === magicNumber) {
                console.log(`Combination N:${combinationCount} (${i} + ${j} = ${magicNumber})`);
                return;
            }
        }
    }
    
    console.log(`${combinationCount} combinations - neither equals ${magicNumber}`);
}